src/common/SurgeSynthesizer.h

```diff
@@ -385,6 +386,19 @@ class alignas(16) SurgeSynthesizer
     void prepareModsourceDoProcess(int scenemask);
     unsigned int saveRaw(void **data);
 
+    // ----  'patch loaded' listener(s) ----
+    // Listeners are notified whenever a patch changes, with the path of the new patch.
+    // Listeners should do any significant work on their own thread; for example, the OSC
+    // patchLoadedListener calls OSCSender::send(), which runs on a juce::MessageManager thread.
+    //
+    // Be sure to delete any added listeners in the desctructor of the class that added them.
+    std::unordered_map<std::string, std::function<void(std::string)>> patchLoadedListeners;
```

Comment:
I guess for same reason we could have made this `std::function<void(const fs::path &)>` but lets leave it for now.

Reply:
Is <void(fs::path)> ok? I've been burned using pointers here.

Reply:
You really don’t want to copy if.  & is wrong but const & is correct